TL/20_Test_Hooks-Before&After/Test_Hooks-Before_&_After_README.md

```diff
@@ -0,0 +1,215 @@
+   scenario and allow you to perform custom actions before and after scenario execution.
+   Around hooks are more flexible than before and after hooks because they can control the entire
+   scenario execution flow.
+
+
+5. BeforeAll and AfterAll hooks are used for same purposes as Before and After in case there is a separation of
+   preconditions required for specific tests divided by some tags (as : UI, API, Regression, Production,
+   Sanity and others), still some general preconditions are required in order to start any test pack.
+
+### Examples
+
+For better understanding, lets visualize some examples:
+
+- simple hooks configuration as a separate class. If Hooks are implemented in a separate class it is
+  recommended to use same package as for step definitions, either to insert the path in the runner class.
+
+```java
+package stepDefinitions;
+
+import io.cucumber.java.After;
+import io.cucumber.java.Before;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.chrome.ChromeDriver;
+
+public class Hooks {
+    public static WebDriver driver;
+
+    @Before
+    public void launchBrowser() {
+        driver = new ChromeDriver();
+        driver.manage().window().maximize();
+    }
+
+    @After
+    public void tearDown() {
+        driver.close();
+        driver.quit();
+    }
+}
+```
+
+- TestRunner class with "hooks" package in glue (path to Hooks class)
+
+```java
+package runner;
+
+
+import io.cucumber.junit.Cucumber;
+import io.cucumber.junit.CucumberOptions;
+import org.junit.runner.RunWith;
+
+@RunWith(Cucumber.class)
+@CucumberOptions(
+        features = "src/test/resources/features",
+        glue = {"stepDefinitions", "hooks"},
+        monochrome = true,
+        plugin = {"pretty"}
+)
+
+public class TestRunner {
+}
+```
+
+***
+
+Let's analyze a more complex example of the Hooks class:
+
+- In this case we have a specific separation of UI and API tests with tags "@UI"/"@API"
+- BeforeAll and AfterAll hooks are used for both test types.
+- AfterStep hook is used to save a screenshot for a detailed UI test evidence
+- Every hook has some information logged, recommended to add scenario name to Before hook.
+
+```java
+package stepDefinitions;
+
+import api.Authorization;
+import io.cucumber.java.*;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.junit.jupiter.api.AfterEach;
+import utils.ScenarioContext;
+import utils.WebDriverFactory;
+
+public class Hooks {
+    private static final Logger logger = LogManager.getLogger(Hooks.class);
+
+    @BeforeAll
+    public static void launchTests() {
+        ScenarioContext.getScenarioInstance().clearContext();
+        logger.info("Test/s started");
+    }
+
+    @Before("@API")
+    public static void setUpAPI(Scenario scenario) {
+        logger.info(System.lineSeparator() + "Starting API test: " + scenario.getName());
+        Authorization.authorization();
+    }
+
+    @Before("@UI")
```

Comment:
add also please some details and examples about order and timeout. That's also some attributes that have Before/After annotations

Reply:
Added a separate chapter about attributes (Tags, order and timeout)